internal/olm/operator/internal/registry_pod_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+
+package olm
+
+import (
+	"context"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+// newFakeClient() returns a clientset
+func newFakeClient() kubernetes.Interface {
+	return fake.NewSimpleClientset()
+}
+
+func TestCreatePod(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Test registry pod")
+}
+
+var _ = Describe("Creation", func() {
+	rp := &RegistryPod{
+		Kubeclient:  newFakeClient(),
+		DBPath:      "/Database/Index.db",
+		BundleImage: "quay.io/joelanford/example-operator-bundle:0.2.0",
+		Namespace:   "default",
+	}
+
+	It("validate registry pod - valid", func() {
+		rp.setDefaults()
+		err := rp.validate()
+
+		Expect(err).To(BeNil())
+	})
+
+	It("create pod", func() {
+		pod, err := rp.Kubeclient.CoreV1().Pods("default").Create(context.Background(),
+			&corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "test-pod"}}, metav1.CreateOptions{})
+
+		Expect(err).To(BeNil())
+		Expect(pod.Name).To(Equal("test-pod"))
+	})
+
+	It("get pod name", func() {
+		podName, err := getPodName(rp.BundleImage)
+		expectedPodName := "example-operator-bundle-index"
+
+		Expect(err).To(BeNil())
+		Expect(podName).To(Equal(expectedPodName))
+	})
+
+	It("valid bundle image name", func() {
+		_, err := rp.build()
+
+		Expect(err).To(BeNil())
```

Comment:
We should add some assertions on the returned pod, based on what `build()` does.